### ISSUE TITLE
v0.23.0 has a TypeError in newline-after-import

### ISSUE THREAD

**Issue by ddolcimascolo** [benmosher/eslint-plugin-import#2056]
Hi guys,

Our build fails when upgrading to v2.23.0 with lots of
`error  Unexpected use of file extension "js" for "./AboutInfo.js" import/extensions`

The rule's actually right, this is something we need to fix. But if it's a change in some defaults, maybe you should have released a major version. WDYT?
Typically minors and patches do not introduce breaking changes. 

Cheers!

David

**Comment by likezero** [comment]
![image](https://user-images.githubusercontent.com/2275857/118239688-b05ac980-b4cc-11eb-8c07-0798a945a68d.png)

**Comment by tgxpuisb** [comment]
as the same

i have locked the version

**Comment by martinstark** [comment]
Can chime in that something is broken in 2.23.0

```
TypeError: Cannot read property 'type' of null
Occurred while linting /home/ms/dev/web-player/cc-sender/src/index.tsx:8
    at isExportNameClass (/home/ms/dev/web-player/node_modules/eslint-plugin-import/lib/rules/newline-after-import.js:51:69)
    at checkForNewLine (/home/ms/dev/web-player/node_modules/eslint-plugin-import/lib/rules/newline-after-import.js:79:45)
    at checkImport (/home/ms/dev/web-player/node_modules/eslint-plugin-import/lib/rules/newline-after-import.js:133:9)
    at /home/ms/dev/web-player/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/home/ms/dev/web-player/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/home/ms/dev/web-player/node_modules/eslint/lib/linter/node-event-generator.js:256:26)
    at NodeEventGenerator.applySelectors (/home/ms/dev/web-player/node_modules/eslint/lib/linter/node-event-generator.js:285:22)
    at NodeEventGenerator.enterNode (/home/ms/dev/web-player/node_modules/eslint/lib/linter/node-event-generator.js:299:14)
    at CodePathAnalyzer.enterNode (/home/ms/dev/web-player/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:711:23)
```

**Comment by nthtrung09it** [comment]
I have same issue :( 

ESLint: 7.26.0
TypeError: Cannot read property 'type' of null

**Comment by starandtina** [comment]
me too....

**Comment by minwe** [comment]
👍

**Comment by mlbiche** [comment]
We also have many false positive `import/no-extraneous-dependencies` and `import/order` in our CI logs since the 2.23.0 release, that we do not have on 2.22.1.

**Comment by SoloJiang** [comment]
We also have something wrong with this version:
![image](https://user-images.githubusercontent.com/14757289/118251235-e3f02080-b4d9-11eb-8841-01b0efe044d4.png)

**Comment by waitingsong** [comment]
same issue

**Comment by pchalupa** [comment]
Same issue

**Comment by joriewong** [comment]
same issue

```bash
Oops! Something went wrong! :(

ESLint: 7.26.0

TypeError: Cannot read property 'type' of null
```

**Comment by slaweet** [comment]
Same issue:
```
$ vue-cli-service lint
 ERROR  TypeError: Cannot read property 'type' of null
Occurred while linting /home/runner/_work/frontend/frontend/src/api/mocks/index.ts:2
TypeError: Cannot read property 'type' of null
Occurred while linting /home/runner/_work/frontend/frontend/src/api/mocks/index.ts:2
    at isExportNameClass (/home/runner/_work/frontend/frontend/node_modules/eslint-plugin-import/lib/rules/newline-after-import.js:51:69)
    at checkForNewLine (/home/runner/_work/frontend/frontend/node_modules/eslint-plugin-import/lib/rules/newline-after-import.js:79:45)
    at checkImport (/home/runner/_work/frontend/frontend/node_modules/eslint-plugin-import/lib/rules/newline-after-import.js:133:9)
    at /home/runner/_work/frontend/frontend/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/home/runner/_work/frontend/frontend/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/home/runner/_work/frontend/frontend/node_modules/eslint/lib/linter/node-event-generator.js:256:26)
    at NodeEventGenerator.applySelectors (/home/runner/_work/frontend/frontend/node_modules/eslint/lib/linter/node-event-generator.js:285:22)
    at NodeEventGenerator.enterNode (/home/runner/_work/frontend/frontend/node_modules/eslint/lib/linter/node-event-generator.js:299:14)
    at CodePathAnalyzer.enterNode (/home/runner/_work/frontend/frontend/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:711:23)
error Command failed with exit code 1.
info Visit yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.
```

**Comment by ljharb** [comment]
Please do not post “same issue” unless you can provide new information. I will fix the type error as soon as i can.

@ddolcimascolo a new eslint warning that should have always appeared is typically not considered a breaking change, otherwise every bugfix would be one. Unfortunately your issue has been hijacked by the type error, so I’ll get to that after publishing a v2.23.1 that fixes the crash.

**Comment by ljharb** [comment]
@SoloJiang please file a new issue for that, so it doesn’t get lost in the noise here.

**Comment by ljharb** [comment]
@martinstark can you provide the content of `/home/runner/_work/frontend/frontend/src/api/mocks/index.ts` so i can make a test case?

**Comment by ljharb** [comment]
I believe the crash should be solved in 8d7ec179 / #2063.

I'll close this for now - @ddolcimascolo, would you mind filing a new issue, and include the code that's warning now?

**Comment by ddolcimascolo** [comment]
@ljharb I could, but it's really any `require('×××.js')` instead of `require('×××')`. I'll fix these in our codebase anyway so I'm good

**Comment by ljharb** [comment]
As long as that's only for relative files (package names ending in ".js" aren't an extension) then that seems like indeed just a bugfix in the rule.

**Comment by ddolcimascolo** [comment]
Yes, only relative files. A few `require('./index.js')` also that should read `require('.')`

**Comment by martinstark** [comment]
@ljharb sorry for the delay! I take it I don't need to provide the code example anymore given this has now been closed

**Comment by ljharb** [comment]
@martinstark as long as it’s fixed for you, then no :-)
